kedro/io/text_local.py

```diff
@@ -118,3 +119,21 @@ def _exists(self) -> bool:
         except DataSetError:
             return False
         return Path(path).is_file()
+
+def _expand_user(filepath):
+    """Return the argument with an initial component of ~ or ~user
```

Comment:
We use google style docstrings, could you please update? See the other files for examples!

Reply:
Since the typehint of `filepath` in the constructor is `str`, we can assume that `self._filepath` will always be a string. Hence you can avoid this function altogether by replacing line 86 with `os.path.expanduser(filepath)` :)